font/src/ft/mod.rs

```diff
@@ -302,32 +294,55 @@ impl FreeTypeRasterizer {
         Ok(font_key)
     }
 
-    fn full_metrics(&self, key: FontKey) -> Result<FullMetrics, Error> {
-        let face = self.faces.get(&key).ok_or(Error::FontNotLoaded)?;
+    fn full_metrics(&self, face_load_props: &FaceLoadingProperties) -> Result<FullMetrics, Error> {
+        let size_metrics =
+            face_load_props.ft_face.size_metrics().ok_or(Error::MissingSizeMetrics)?;
 
-        let size_metrics = face.ft_face.size_metrics().ok_or(Error::MissingSizeMetrics)?;
-
-        let width = match face.ft_face.load_char('0' as usize, face.load_flags) {
-            Ok(_) => face.ft_face.glyph().metrics().horiAdvance / 64,
-            Err(_) => size_metrics.max_advance / 64,
-        } as f64;
+        let width =
+            match face_load_props.ft_face.load_char('0' as usize, face_load_props.load_flags) {
+                Ok(_) => face_load_props.ft_face.glyph().metrics().horiAdvance / 64,
+                Err(_) => size_metrics.max_advance / 64,
+            } as f64;
```

Comment:
Might be worth to extract `face_load_props.ft_face` into a variable here, since it would prevent two very ugly linebreaks.